### PR TITLE
Pre-Election Reporting Dates Template

### LINKED ISSUE #3016: Design a new table in Wagtail to display the 2020 congressional pre-primary reports info
### Summary

_Looking to create a table in Wagtail that is user friendly and easy to update the 2020 reporting information. The table will be used on the child pages to display reporting information under Dates and deadlines._

### Proposed Behavior / Feature

- [x] Looking to create a dropdown for an individual to pick a state to view reporting information.
- [x] Looking for an option to display all 50 states and territories, so the regulated community can also see the big picture. 
- [x] Decrease possibility of human error by creating a table that is easy to input data and display information clearly using Wagtail.
- [x] Choose tool we'll use to populate data.

### Current Situation

_Current situation is we are using straight html code in the template. The current table for pre-election report information is huge. The increased ability for human error is a factor, and it is very time consuming to update._

### Use Case

_The content team/Info division would benefit from this added feature along with the regulated community._

### Misc

Here is a link to how the data was displayed in the 2018 election 
https://transition.fec.gov/info/charts_primary_dates_2018.shtml

A draft template is in Wagtail under " 2020 Dates and deadlines." It is a child page under this heading, which will display the lengthy html code when a chart is inserted as html code. 

### Future Work

_Other tables to be created in #3017_

### PR DESCRIPTION
**Pull request by johnnyporkchops**
## Summary (required)
This is the full-width page Wagtail template and accompanying JS that will allow us to achieve the mocked up experience and responsive design in issue #3016 
- Resolves #3016
_Include a summary of proposed changes._

Related PR: Closing #3188 in favor of this PR

## Impacted areas of the application
modified:   fec/static/scss/components/_table-styles.scss
modified:   home/models.py
modified:   ../fec/static/js/polyfills.js
modified:   ../package-lock.json
modified:   ../package.json
modified:   fec/templates/base.html

new: fec/static/js/pages/reporting-dates-tables.js
new: home/migrations/0105_auto_20191002_1534.py
new: home/templates/home/full_width_page.html
new: home/migrations/0105_auto_20191002_1534

**To Do:** (All but the first one here are easy fixes)
- [x] Resolve issue with using accordion on this page. Accordion does not work for the `All footnotes` section. `data-init.js` (required for the A11Y dialog) interferes with the accordion trigger. I have left the native `HTML5 Details` element here for now.
- [x] Tweak any styling or design  changes requested by @Jonella
- [x] Clean up JS, review comments for brevity and clarity 
- [x] Decide the cell-border color that works for both white and zebra-striped rows
- [x] switch hex colors in SCSS with proper variables (ie: `#ddd` becomes `$gray-medium` or whatever)
- [x]  SCSS - Remove as many `!important` rules as possible - some may not be necessary

## Screenshots
Screenshots to come soon

## How to test
1) Checkout `feature/3016-reporting-dates-table-template`
**Steps 2 through 6 are designed to keep your local `cfdm_cms_test` database in tact for working with other branches. You can forego this and just test this [on feature](https://fec-feature-cms.app.cloud.gov/new-test-pre-election-tables-responsive/).**
*******************
2) `craetedb cfdm_cms_test_2019`
3) Get the latest database dump from [Google drive](https://drive.google.com/drive/folders/0B6ktRyka2SlDaC1hb1k3aHA5Zms)
4) `pg_restore --dbname cfdm_cms_test_2019 --no-acl  --no-owner <LOCAL PATH TO DUMP FILE NAME> `
5) `export DATABASE_URL=postgresql://:@/cfdm_cms_test_2019` (see step 14 to point back)
6) `./manage.py migrate`
***************************
7) CD to /fec-cms and `rm -rf node_modules`
8)  `npm install`
9) `npm run build`
10) cd to fec-cms/fec/ and ` ./manage.py runserver`
11) go to `127.0.0.1/admin`
12) create a page of type `FullWidthPage`
13) create three html blocks and paste in the sections from the  attached  files into the three blocks. This step simulates the content team members copy/pasting from Excel but leaves out the actual spreadsheet part here.
[sample-data.txt](https://github.com/fecgov/fec-cms/files/3637508/sample-data.txt)

14) When you are done, point your local database env var back to export cfdm_cms_test:
`export DATABASE_URL=postgresql://:@/cfdm_cms_test` or if you use a bash profile with your database env var saved, just run:
`unset DATABASE_URL`

### CODE REVIEW

File: fec/home/templates/home/full_width_page.html [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+{% block body_class %}template-{{ self.get_verbose_name | slugify }}{% endblock %}
+<script type="text/javascript" src="{% asset_for_js 'reporting-dates-tablesIE2.js' %}"></script>
+{% block content %}
+
+{% block breadcrumbs %}
+  {% include 'partials/breadcrumbs.html' with page=self links=self.get_ancestors style=self.breadcrumb_style %}
+{% endblock %}
+
+<article class="main container">
+
```
Comment (patphongs) [inline fec/home/templates/home/full_width_page.html]: Can you add the main title in here? `<h1 class="heading--main">{{ self.title }}</h1>`

File: fec/fec/static/scss/components/_table-styles.scss [hunk 2]
```diff
@@ -151,6 +151,264 @@ h3 + .simple-table {
...
+        display: none !important;
+      }
+    }
+
+     td {
+      /* Behave  like a "row" */
+      border: none;
+      border-bottom: 1px solid $gray-light !important;
+      position: relative;
+      font-weight: bold;
```
Comment (patphongs) [inline fec/fec/static/scss/components/_table-styles.scss]: Should this be removed? It's making the whole row bold in mobile and I'm assuming we only want the `.th_append` to be bold in mobile?

Screen Shot 2019-10-03 at 9 56 34 PM

Comment (johnnyporkchops) [reply fec/fec/static/scss/components/_table-styles.scss]: According to the mockup, I think they are supposed to be bold
https://user-images.githubusercontent.com/31663028/64990128-27c1fb00-d89d-11e9-8d89-57055218bbb5.png

Comment (JonellaCulmer) [reply fec/fec/static/scss/components/_table-styles.scss]: I think we only want the column names to be bold, not the dates.

**Comment by johnnyporkchops** [comment]
I think we should consider two page types for full-width page. 
1) generic `full-width` page
2) `full-width-table` page( which would start out as being just for the reporting dates tables ) and eventually become a template for any tables we want to use the new mobile view option.
This way, if one wants to create a generic full width page without the dates tables, it will not include all of the javascript necessary to make the dates tables work.

**Comment by patphongs** [comment]
> I think we should consider two page types for full-width page.

I agree, we should have two in a separate PR.

File: fec/fec/static/scss/components/_table-styles.scss [hunk 3]
```diff
@@ -151,6 +151,241 @@ h3 + .simple-table {
...
+      border-bottom: none;
+    }
+  }
+}
+
+.election-dates-table {
+  tr {
+    display: table-row ;
+    &.row_display {
+      display: table-row;
```
Comment (rfultz) [inline fec/fec/static/scss/components/_table-styles.scss]: It isn't breaking anything but do we need this twice?

File: fec/fec/static/scss/components/_table-styles.scss [hunk 4]
```diff
@@ -151,6 +151,241 @@ h3 + .simple-table {
...
+    }
+  }
+
+  td {
+    padding: u(1rem);
+    vertical-align: top;
+    border: 1px solid $gray-light;
+
+      &:first-child {
+        border-left:none;
```
Comment (rfultz) [inline fec/fec/static/scss/components/_table-styles.scss]: What does the linter say about spaces after colons? It only affects anything when we're searching the code looking for something specific, like "border: "—really minor thing.

File: fec/fec/static/scss/components/_table-styles.scss [hunk 5]
```diff
@@ -151,6 +151,241 @@ h3 + .simple-table {
...
+    height: u(1.7rem);
+    margin-left: u(.2rem);
+  }
+
+  td a[href^='#'], th a[href^='#'] {
+    border-bottom: none;
+  }
+
+  tr.footnote_row {
+    /*background: $inverse !important;*/
```
Comment (rfultz) [inline fec/fec/static/scss/components/_table-styles.scss]: Can we remove the comments of unused code? (Or explain why we want them? (I know I add and then forget too many comments))

File: fec/fec/templates/base.html [hunk 6]
```diff
@@ -173,6 +173,15 @@ <h2 style="color: #ffffff">Your browser is outdated</h2>
...
     <script defer type="text/javascript" src='https://www.google.com/recaptcha/api.js'></script>
+    <script>
+      var canSkipPolyfills = ('fetch' in window && 'assign' in Object);
+      if (!canSkipPolyfills) {
+        var pfScriptElem = document.createElement('script');
+        pfScriptElem.async = false;
+        pfScriptElem.src = "{% asset_for_js 'polyfills.js' %}";
+        document.head.appendChild(pfScriptElem);
+      }
+    </script>
```
Comment (rfultz) [inline fec/fec/templates/base.html]: Thank you! Sorry I forgot that one.

**Pull request merged**